packages/gatsby/src/utils/webpack-utils.js

```diff
@@ -115,7 +115,13 @@ module.exports = async ({
   stage: Stage,
   program: any,
 }): Promise<WebpackUtilsOptions> => {
-  const assetRelativeRoot = `static/`
+  /**
+   *  the leading `../` for assetRelativeRoot is required to ensure
+   * the static files extracted by webpack are not part of the
+   * "build-javascript" output path of `/js`. The same technique is
+   * used for CSS
+   */
+  const assetRelativeRoot = `../static/`
```

Comment:
@m-allanson This had to be treated the same way as the `../css/` case because of how there's only a `build-javascript` stage. https://github.com/gatsbyjs/gatsby/pull/6346#discussion_r201193282



Here are all the build stages:
```javascript
type Stage = "develop" | "develop-html" | "build-javascript" | "build-html"
```

I think in the future it would be a good idea to add discrete build stages for CSS and static assets instead of relying on this current workaround.

Here's an example of the output folder of your example repository with this commit

![screen shot 2018-07-12 at 7 00 45 pm](https://user-images.githubusercontent.com/8007686/42668565-311da270-8606-11e8-9fdd-9060a2374b35.png)